app/models/spree/stock_item_decorator.rb

```diff
@@ -0,0 +1,3 @@
+Spree::StockItem.class_eval do
+  validates :count_on_hand, numericality: { greater_than_or_equal_to: 0, unless: :backorderable? }
+end
```

Comment:
now that we are going "bye bye spree" you can bring the full class from spree in these cases :heart: 
If you are up for it you can even bring a few more related things (that way we can save one iteration on the manual testing) like stock_movement and stock_location.
reference: https://github.com/openfoodfoundation/openfoodnetwork/issues/4826#issuecomment-611698328

Reply:
Thank you for this, @luisramos0! For now, I just brought in `Spree::StockItem`.